lib/puppet/provider/package/puppetserver_gem.rb

```diff
@@ -36,9 +36,11 @@ def self.gemlist(options)
     end
 
     begin
+      # when `/tmp` mounted `noexec`, `puppetserver gem list` will output *** LOCAL GEMS ***
```

Comment:
Instead, could gemsplit in ..

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/package/gem.rb#L52

... skip these lines, which would address the issue for the gem, puppet_gem, and puppetserver_gem providers?

Reply:
It could.. but this would need a the fix to be released in an updated puppet agent. Since this bug relates to interaction between `/tmp` being mounted `noexec` and jruby, I don't believe its an issue for the other providers given the comment in the file

Reply:
Thanks, I see that now. 

I do have a PR (https://github.com/puppetlabs/puppetlabs-puppetserver_gem/pull/24) related to https://tickets.puppetlabs.com/browse/ENTERPRISE-1179 that intersects this: https://github.com/puppetlabs/puppetlabs-puppetserver_gem/pull/23 and https://github.com/puppetlabs/puppetlabs-puppetserver_gem/pull/18

What do you think?